samples/client.c

```diff
@@ -753,61 +834,34 @@ static size_t
 do_migrate(int sock, size_t nr_iters, struct iovec *migr_iter)
 {
     int ret;
-    uint64_t pending_bytes, data_offset, data_size;
+    bool is_more = true;
     size_t i = 0;
 
     assert(nr_iters > 0);
 
-    /* XXX read pending_bytes */
-    ret = access_region(sock, VFU_PCI_DEV_MIGR_REGION_IDX, false,
-                        offsetof(struct vfio_user_migration_info, pending_bytes),
-                        &pending_bytes, sizeof(pending_bytes));
-    if (ret < 0) {
-        err(EXIT_FAILURE, "failed to read pending_bytes");
-    }
-
-    for (i = 0; i < nr_iters && pending_bytes > 0; i++) {
-
-        /* XXX read data_offset and data_size */
-        ret = access_region(sock, VFU_PCI_DEV_MIGR_REGION_IDX, false,
-                            offsetof(struct vfio_user_migration_info, data_offset),
-                            &data_offset, sizeof(data_offset));
-        if (ret < 0) {
-            err(EXIT_FAILURE, "failed to read data_offset");
-        }
+    for (i = 0; i < nr_iters && is_more; i++) {
 
-        ret = access_region(sock, VFU_PCI_DEV_MIGR_REGION_IDX, false,
-                            offsetof(struct vfio_user_migration_info, data_size),
-                            &data_size, sizeof(data_size));
-        if (ret < 0) {
-            err(EXIT_FAILURE, "failed to read data_size");
-        }
+        migr_iter[i].iov_len = 1024;
```

Comment:
Why exactly 1K? Generally page size is preferrable. You can allocate outside this loop once and reuse the buffer. Need to check for memory allocation failures.

Reply:
This has been since resolved.